modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,764 @@
+    static void run(const cv::Mat &in_mat,
+        cv::Mat &out_mat) {
+        cv::transpose(in_mat, out_mat);
+    }
+};// GAPI_OCV_KERNEL(Transpose)
+} // anonymous namespace
+} // namespace custom
+
+namespace vis {
+namespace {
+void bbox(cv::Mat& m, const cv::Rect& rc) {
+    cv::rectangle(m, rc, cv::Scalar{ 0,255,0 }, 2, cv::LINE_8, 0);
+};
+
+using rectPoints = std::pair<cv::Rect, std::vector<cv::Point>>;
+
+static cv::Mat drawRectsAndPoints(const cv::Mat& img,
+    const std::vector<rectPoints> data) {
+    cv::Mat outImg;
+    img.copyTo(outImg);
+
+    for (auto& d : data) {
+        vis::bbox(outImg, d.first);
+        auto pts = d.second;
+        for (size_t i = 0; i < pts.size(); ++i) {
+            cv::circle(outImg, pts[i], 3, cv::Scalar(0, 255, 255), 1);
+        }
+    }
+    return outImg;
+}
+} // anonymous namespace
+} // namespace vis
+
+
+//Infer helper function
+namespace {
+static inline std::tuple<cv::GMat, cv::GMat> run_mtcnn_p(cv::GMat &in, const std::string &id) {
+    cv::GInferInputs inputs;
+    inputs["data"] = in;
+    auto outputs = cv::gapi::infer<cv::gapi::Generic>(id, inputs);
+    auto regressions = outputs.at("conv4-2");
+    auto scores = outputs.at("prob1");
+    return std::make_tuple(regressions, scores);
+}
+
+//Operator fot PNet network package creation in the loop
+inline cv::gapi::GNetPackage& operator += (cv::gapi::GNetPackage& lhs, const cv::gapi::GNetPackage& rhs) {
+    lhs.networks.reserve(lhs.networks.size() + rhs.networks.size());
+    lhs.networks.insert(lhs.networks.end(), rhs.networks.begin(), rhs.networks.end());
+    return lhs;
+}
+
+static inline std::string get_pnet_level_name(const cv::Size &in_size) {
+    return "MTCNNProposal_" + std::to_string(in_size.width) + "x" + std::to_string(in_size.height);
+}
+
+int calculate_scales(const cv::Size &input_size, std::vector<double> &out_scales, std::vector<cv::Size> &out_sizes ) {
+    //calculate multi - scale and limit the maxinum side to 1000
+    //pr_scale: limit the maxinum side to 1000, < 1.0
+    double pr_scale = 1.0;
+    double h = static_cast<double>(input_size.height);
+    double w = static_cast<double>(input_size.width);
+    if (std::min(w, h) > 1000)
+    {
+        pr_scale = 1000.0 / std::min(h, w);
+        w = w * pr_scale;
+        h = h * pr_scale;
+    }
+    else if (std::max(w, h) < 1000)
+    {
+        w = w * pr_scale;
+        h = h * pr_scale;
+    }
+    //multi - scale
+    out_scales.clear();
+    out_sizes.clear();
+    double factor = 0.709;
+    int factor_count = 0;
+    double minl = std::min(h, w);
+    while (minl >= 12)
+    {
+        double current_scale = pr_scale * std::pow(factor, factor_count);
+        cv::Size current_size(static_cast<int>(static_cast<double>(input_size.width) * current_scale),
+                              static_cast<int>(static_cast<double>(input_size.height) * current_scale));
+        std::cout << "current_scale " << current_scale << std::endl;
```

Comment:
Is it a debug print or sample info?

Reply:
sample info to be sure that blob format is correct for input image size